src/timestamp.rs

```diff
@@ -124,4 +142,26 @@ impl Interval {
     pub fn lerp(self, value: f32) -> Timestamp {
         Timestamp((value * (self.duration_ns() as f32)).round() as i64 + self.start.0)
     }
+
+    // convert a string like "500.0 s" to a timestamp
+    pub fn convert_str_to_timestamp(s: &str) -> Result<Timestamp, IntervalParseError> {
+        let mut parts = s.split_whitespace();
+        let value = parts.next().ok_or(IntervalParseError::NoValue)?;
+        let value = value
+            .parse::<f32>()
+            .map_err(|_| IntervalParseError::InvalidValue)?;
+        let unit = parts.next().ok_or(IntervalParseError::NoUnit)?;
+        let unit = unit.to_lowercase();
+        let ns_per_us = 1_000;
+        let ns_per_ms = 1_000_000;
+        let ns_per_s = 1_000_000_000;
+        let ns = match unit.as_str() {
+            "ns" => value as i64,
+            "us" => (value * ns_per_us as f32) as i64,
+            "ms" => (value * ns_per_ms as f32) as i64,
+            "s" => (value * ns_per_s as f32) as i64,
+            _ => return Err(IntervalParseError::InvalidUnit),
+        };
+        Ok(Timestamp(ns))
```

Comment:
Some general thoughts on the parser:

 * If I'm not mistaken, we don't ever check `parts` again, so something like `1.0 s asdf` will parse despite having garbage at the end.
 * It seems reasonable to write `1.0s` (no space) but currently that wouldn't parse.
 * Please write unit tests for this function, it is small, self-contained and (importantly) tricky enough we could easily miss a corner case.